src/devices/cpu/uml.cpp

```diff
@@ -520,13 +537,26 @@ void uml::instruction::simplify()
 				else if (m_param[2].is_immediate() && m_param[3].is_immediate())
 				{
 					if (m_size == 4)
-						convert_to_mov_immediate(s32(s32(m_param[1].immediate()) * s32(m_param[2].immediate())));
+						convert_to_mov_immediate(s32(s32(m_param[2].immediate()) * s32(m_param[3].immediate())));
 					else if (m_size == 8)
-						convert_to_mov_immediate(s64(s64(m_param[1].immediate()) * s64(m_param[2].immediate())));
+						convert_to_mov_immediate(s64(s64(m_param[2].immediate()) * s64(m_param[3].immediate())));
 				}
 			}
 			break;
 
+		// MULSLW: convert simple form to MOV if immediate, or if multiplying by 0
+		case OP_MULSLW:
+			if (m_param[1].is_immediate_value(0) || m_param[2].is_immediate_value(0))
+				convert_to_mov_immediate(0);
+			else if (m_param[1].is_immediate() && m_param[2].is_immediate())
+			{
+				if (m_size == 4)
+					convert_to_mov_immediate(s32(s32(m_param[1].immediate()) * s32(m_param[2].immediate())));
+				else if (m_size == 8)
+					convert_to_mov_immediate(s64(s64(m_param[1].immediate()) * s64(m_param[2].immediate())));
+			}
+			break;
```

Comment:
Same here – will the overflow flag be updated correctly by the simplified version?

Reply:
Same as above comment so I'll resolve this comment and leave the other for discussion.